code/mccSearch.py

```diff
@@ -542,7 +612,838 @@ def find_cloud_elements(mergImgs, timelist, mainStrDir, lat, lon, TRMMdirName=No
+                
+                precipTotal = np.sum(finalCETRMMvalues)
+                #assert(precipTotal==np.sum(finalCETRMMvalues)) #mini test
+                
+                start = time.time()
+                #TODO: this step is really slow. We could probably get away with storing
+                # stuff in a different format which goes faster
+                rainFallacc[:] = finalCETRMMvalues[:]
+                currNetCDFTRMMData.close()
+                TRMMnumOfBoxes = np.count_nonzero(finalCETRMMvalues)
+                TRMMArea = TRMMnumOfBoxes * XRES * YRES
+                P_TIME += time.time() - start
+                
+                try:
+                    maxCEprecipRate = np.max(finalCETRMMvalues[np.nonzero(finalCETRMMvalues)])
+                    minCEprecipRate = np.min(finalCETRMMvalues[np.nonzero(finalCETRMMvalues)])
+                except:
+                    pass
+                
+                
+
+            profTimes[7]+= time.time() - pin
+            pin = time.time() #ninth pin drop
+
+            #sort cloudElementLatLons by lats
+            cloudElementLatLons.sort(key=lambda tup: tup[0])
+
+            #determine if the cloud element the shape
+            cloudElementEpsilon = eccentricity(cloudElement)
+            cloudElementsUserFileString+=('\n\nTime is: %s' %(str(timelist[t])))
+            cloudElementsUserFileString+=('\nceUniqueID is: %s' %ceUniqueID)
+            latCenter, lonCenter = ndimage.measurements.center_of_mass(cloudElement, labels=labels)
+            #latCenter and lonCenter are given according to the particular array defining this CE
+            #so you need to convert this value to the overall domain truth
+            latCenter = cloudElementLat[round(latCenter)]
+            lonCenter = cloudElementLon[round(lonCenter)]
+            
+            #NEED TO FIX-----------------------------------------------------------------
+            #create the latLonBox
+            latLonBox.append(min(cloudElementLon))
+            latLonBox.append(min(cloudElementLat))
+            latLonBox.append(max(cloudElementLon))
+            latLonBox.append(max(cloudElementLat))
+
+            #for _ in cloudElement:
+            #    #assign a matrix to determine the legit values
+            #    nonEmptyLons = sum(sum(cloudElement) > 0)
+            #    nonEmptyLats = sum(sum(cloudElement.transpose()) > 0)
+                
+            nonEmptyLons = sum(sum(cloudElement) > 0)
+            nonEmptyLats = sum(sum(cloudElement.transpose()) > 0)
+
+            shape = max(nonEmptyLats, nonEmptyLons)
+            
+            cloudElementsUserFileString+=('\nCenter (lat,lon) is: %.2f\t%.2f' %(latCenter, lonCenter))
+            cloudElementCenter.append(latCenter)
+            cloudElementCenter.append(lonCenter)
+            cloudElementsUserFileString+=('\nNumber of boxes are: %d' %numOfBoxes)
+            cloudElementsUserFileString+=('\nArea is: %.4f km^2' %(cloudElementArea))
```

Comment:
Hey @kwhitehall and @meldefon the best approach would be to define the string once all the input variable values are known.  You can easily build a multi-line string and call the .format() built-in function.

Here is a good cookbook I found online:
https://mkaz.github.io/2012/10/10/python-string-format/

The python docs on this are a bit too cryptic and dense for me.

Example:  
"""Number of boxes are: {boxCount:d}
Area is: {cloudArea:.4f} km^2""".format(boxCount= numOfBoxes, cloudArea= cloudElementArea)